starter-core/src/test/groovy/io/micronaut/starter/feature/FeatureSpec.groovy

```diff
@@ -36,23 +39,28 @@ class FeatureSpec extends BeanContextSpec {
     @Unroll
     void "test #feature does not add an unmodifiable map to config"(Feature feature) {
         when:
-        JdkVersion javaVersion = javaVersionForFeature(feature.getName())
+        ApplicationType applicationType = applicationTypeForFeature(feature.name)
+        JdkVersion javaVersion = javaVersionForFeature(feature.name)
         Language language = Language.JAVA
         if (feature instanceof LanguageSpecificFeature) {
             language = ((LanguageSpecificFeature) feature).getRequiredLanguage()
         }
         Options options = new Options(language, TestFramework.JUNIT, BuildTool.GRADLE, javaVersion)
-        def features = [feature.getName()]
+        List<String> features = [feature.getName()]
+
         if (feature instanceof JAsyncSQLFeature) {
             // JAsyncSQLFeatureValidator fails unless exactly one of mysql or postgress are included
             // so it can't be tested in isolation like this in isolation
             features += 'mysql'
+        } else if (feature instanceof Cdk) {
+            // Cdk fails unless it is combined with Lambda
+            features += AwsLambda.FEATURE_NAME_AWS_LAMBDA
```

Comment:
`+=` creates a new instance and copies old entries and adds the new one(s) - just use `<<` or `.add()`